File: Selection_Sort.cpp
#include <iostream>
#include <algorithm>
using namespace std;

void selectionSort(int arr[], int n)
{
    for(int i=0;i<n-1;i++)
    {
        int minIdx=i;
        for(int j=minIdx+1;j<n;j++)
        {
            if(arr[j]<arr[minIdx])
                minIdx=j;
        }
        swap(arr[i],arr[minIdx]);
    }
}

int main() {

    int arr[] {99,10,9,8,7,6,5,4,3,2,1,99};
    int n=sizeof(arr)/sizeof(arr[0]);
    selectionSort(arr,n);
    for(int i=0;i<n;i++)
        cout<<arr[i]<<",";
    return 0;
}

File: python codes/Sorting Algorithms/selection_sort.py
import sys

def selection_sort(arr):
    for i in range(len(arr)):
        min_indx = i

        for j in range(i+1, len(arr)):
            if arr[min_indx] > arr[j]:
                min_indx = j
        
        arr[i], arr[min_indx] = arr[min_indx], arr[i]

    return arr


arr = [0, 8, 64, 5, 87, 3, 54]
print(selection_sort(arr))